crates/swc_ecma_minifier/tests/terser/compress/switch/drop_case_2/output.js

```diff
@@ -0,0 +1 @@
+if ("bar" === foo) bar();
```

Comment:
This would require a context aware side effect checker -- `swc` haven't got one yet

Reply:
Do you mean we need one for `foo`?
We do not support side effects on identifiers.

We should check if it's declared, though.
But I think we are already doing so, aren't we?

Reply:
> We should check if it's declared, though.
> But I think we are already doing so, aren't we?

No, `may_have_side_effects` is on `ExprExt` which have no information about declared or not